lntest/itest/lnd_multi-hop-error-propagation.go

```diff
@@ -382,19 +390,28 @@ out:
 		t.Fatalf("unable to reset mission control: %v", err)
 	}
 
-	sendReq = &lnrpc.SendRequest{
-		PaymentRequest: carolInvoice.PaymentRequest,
-	}
 	ctxt, _ = context.WithTimeout(ctxb, defaultTimeout)
-	resp, err = net.Alice.SendPaymentSync(ctxt, sendReq)
+	stream, err = net.Alice.RouterClient.SendPaymentV2(
+		ctxt,
+		&routerrpc.SendPaymentRequest{
+			PaymentRequest: carolInvoice.PaymentRequest,
+			TimeoutSeconds: 60,
+			FeeLimitMsat:   noFeeLimitMsat,
+		},
+	)
 	if err != nil {
 		t.Fatalf("unable to send payment: %v", err)
 	}
-
-	if resp.PaymentError == "" {
-		t.Fatalf("payment should have failed")
+	resp, err = getPaymentResult(stream)
+	if err != nil {
+		t.Fatalf("cannot get payment result: %v", err)
 	}
+	if resp.Status != lnrpc.Payment_FAILED ||
```

Comment:
We could refactor `sendAndAssertSuccess` to be a bit more generic and take the expected response status and reason.

Reply:
Create a similar send and assert method for failures